retriever/lib/socrata.py

```diff
@@ -0,0 +1,260 @@
+import os
+import json
+
+import requests
+from tqdm import tqdm
+from urllib.error import HTTPError
+
+from retriever.lib.create_scripts import create_package
+from retriever.lib.defaults import SOCRATA_BASE_URL, SOCRATA_SCRIPT_WRITE_PATH
+from retriever.lib.scripts import reload_scripts
+from retriever.lib.templates import BasicTextTemplate
+
+
+def socrata_autocomplete_search(dataset):
+    """Returns the list of dataset names after autocompletion"""
+    query = " ".join(dataset)
+    url = SOCRATA_BASE_URL + '/autocomplete'
+    params = {'q': query}
+    try:
+        response = requests.get(
+            url,
+            params,
+            headers={
+                'user-agent':
+                    'Weecology/Data-Retriever Package Manager: http://www.data-retriever.org/'
+            },
+            allow_redirects=True,
+            stream=True)
+
+        if response.status_code == 404:
+            print("Socrata is unable to fetch dataset names currently")
+            return []
+        else:
+            datasets = response.json()
+            names = []
+
+            for i in range(len(datasets["results"])):
+                names.append(datasets["results"][i]["title"])
+
+    except HTTPError as e:
+        print("HTTPError :", e)
+        return []
```

Comment:
Use return `None` or  `False`. In this case if we return None on the list variable, we can use `if name` to check if its empty.